compiler/src/iree/compiler/GlobalOptimization/PropagateLinalgTranspose.cpp

```diff
@@ -440,6 +674,56 @@ class PropagateTransposeThroughUnaryElementwiseInput
   }
 };
 
+// Propagates a transpose through the input of a unary elementwise operation.
+class PropagateTransposeThroughUnaryElementwiseDpsInit
+    : public OpRewritePattern<linalg::TransposeOp> {
+public:
+  using OpRewritePattern<linalg::TransposeOp>::OpRewritePattern;
+
+  LogicalResult matchAndRewrite(linalg::TransposeOp transposeOp,
+                                PatternRewriter &rewriter) const override {
+    auto genericOp = transposeOp.getInput().getDefiningOp<linalg::GenericOp>();
```

Comment:
A lot of this seems similar to the `DpsInput` case. Why is this separate? Can they be merged?